test/jasmine-test/spec/coreTests.js

```diff
@@ -724,8 +748,43 @@ var coreTests = [
         context:  { foo: {bar: "Hello!"} },
         expected: "Hello!",
         message: "should test an object path"
-      }
-    ]
+      },
+	  {
+		name:     "thenable reference",
+		source:   "Eventually {magic}!",
```

Comment:
what would happen if someone tries to reference `magic.foo`. I remember we had some logic that would fire the magic function and look for foo in it's return value.